### DISCUSSION TITLE
Acaba l'sprint 3 i comença l'sprint 4.

### DISCUSSION THREAD

**Discussion by FherStk** [ElPuig/IMS#7] [1. IMS: general]
Bon dia a tothom,

Perdoneu el retard (això ja sembla el meu eslogan), però l'sprint 3 (temps per a preparar les branques i muntar una proposta d'esquelet de mòdul) va acabar el dia 30 i el dia 1 va començar l'sprint 4. Com ja sabeu, hem definit els sprints perquè tinguin una durada d'un mes i durant aquest quart sprint ens toca el següent:

El Chus ha muntat un esquelet d'un mòdul inicial i ens ha fet una proposta dels models de dades que necessitem. Es tracta que, entre tots, els hi fem una ullada i fem propostes de millora o modificacions. 

El que hauria de quedar clar en acabar l'sprint és:
- Si tenim o no totes les entitats necessàries.
- Si les entitats necessàries tenen o no tots els atributs necessaris.
- Que estiguem d'acord en els noms de les entitats i els seus atributs (proposo fer servir només l'anglès i amb paraules prou genèriques per a facilitar la col·laboració internacional amb qualsevol desenvolupador/a).
- Que l'esquelet que tenim, amb les funcionalitats actuals, funciona com cal i no dona cap error (permisos, etc.).
- _Opcional_: revisar com funciona el sistema de traduccions d'Odoo i tenir el codi predeterminat en anglès però ja traduït al Català. 

Salut i gràcies a tots,
Fer

**Comment by vcarceler** [comment]
Всем привет!

En IsardVDI suben las locales a https://hosted.weblate.org/projects/isardvdi/frontend/. Allí les traduje el frontend al ruso y ahora veo que deben haber añadido cadenas porque ya no está traducido al 100%.

Si me pasáis las locales las subo a esta plataforma y me encargo del ruso y castellano. O si hay otra plataforma favorita pues en la que me digáis :-)

No sé si Odoo utiliza su propio sistema de locales o el genérico de Django: https://docs.djangoproject.com/en/4.2/topics/i18n/translation/

**Reply by FherStk** [reply]
Muchas gracias Victor,

He encontrado esto en la documentación de Odoo: https://www.odoo.com/documentation/16.0/developer/howtos/translations.html

Cuando tenga un hueco, exporto el fichero de traducción y miramos de hacer un par de pruebas.

Saludos,
Fer

**Comment by FherStk** [comment]
Bona tarda,

Acabo de penjar una nova versió del mòdul amb els següents canvis:
- Ja no hi ha warnings a l'hora d'instal·lar o actualitzat. He deixat el codi deprecated comentat.
- He canviat els noms dels models i els seus atributs perquè estiguin tots en anglès. 

Queda pendent:
- Canviar els noms dels literals de les vistes XML.
- Fer proves amb el sistema de traduccions, perquè tinguem sempre l'IMS en anglès i català.
- Comprovar que tots els models disposen dels atributs que necessitem. Tampoc passa res per afegir-ne més endavant, el problema està a esborrar (es pot, però a Odoo no li agrada gaire, mai esborra res com a bon ERP que és). 

Salut,
Fer
